# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/DOMLocalization.cpp
# Commit: 3404fd6725f1
# Full Hash: 3404fd6725f15d1253a7f76e3937185bd353f263
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part2: Switch Localization class to use localization-ffi. r=emilio,nika
#   
#   Depends on D104788
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104789
# ==============================================================================

diff -r c9cfbba15d90 -r 3404fd6725f1 dom/l10n/DOMLocalization.cpp
--- a/dom/l10n/DOMLocalization.cpp	Mon Aug 02 09:54:07 2021 +0000
+++ b/dom/l10n/DOMLocalization.cpp	Mon Aug 02 09:54:07 2021 +0000
@@ -42,14 +42,12 @@
   RefPtr<DOMLocalization> domLoc =
       new DOMLocalization(aGlobal, aSync, aBundleGenerator);
 
-  domLoc->Init();
-
   return domLoc.forget();
 }
 
 DOMLocalization::DOMLocalization(nsIGlobalObject* aGlobal, const bool aSync,
                                  const BundleGenerator& aBundleGenerator)
-    : Localization(aGlobal, aSync, aBundleGenerator) {
+    : Localization(aGlobal, aSync) {
   mMutations = new L10nMutations(this);
 }
 
@@ -70,8 +68,6 @@
     domLoc->AddResourceIds(aResourceIds);
   }
 
-  domLoc->Activate(true);
-
   return domLoc.forget();
 }
 
@@ -302,9 +298,6 @@
     return nullptr;
   }
 
-  AutoEntryScript aes(mGlobal, "DOMLocalization TranslateElements");
-  JSContext* cx = aes.cx();
-
   for (auto& domElement : aElements) {
     if (!domElement->HasAttr(kNameSpaceID_None, nsGkAtoms::datal10nid)) {
       continue;
@@ -335,7 +328,7 @@
   if (mIsSync) {
     nsTArray<Nullable<L10nMessage>> l10nMessages;
 
-    FormatMessagesSync(cx, l10nKeys, l10nMessages, aRv);
+    FormatMessagesSync(l10nKeys, l10nMessages, aRv);
 
     bool allTranslated =
         ApplyTranslations(domElements, l10nMessages, aProto, aRv);
@@ -346,7 +339,7 @@
 
     promise->MaybeResolveWithUndefined();
   } else {
-    RefPtr<Promise> callbackResult = FormatMessages(cx, l10nKeys, aRv);
+    RefPtr<Promise> callbackResult = FormatMessages(l10nKeys, aRv);
     if (NS_WARN_IF(aRv.Failed())) {
       return nullptr;
     }
@@ -528,10 +521,7 @@
 
 void DOMLocalization::OnChange() {
   Localization::OnChange();
-  if (mLocalization && !mResourceIds.IsEmpty()) {
-    ErrorResult rv;
-    RefPtr<Promise> promise = TranslateRoots(rv);
-  }
+  RefPtr<Promise> promise = TranslateRoots(IgnoreErrors());
 }
 
 void DOMLocalization::DisconnectMutations() {